### ISSUE TITLE
Request: Utilize spells_us.txt for server spell parsing of spells/abilities

### ISSUE THREAD

**Issue by AbyssalStalker** [kauffman12/EQLogParser#27]
Howdy,

I recently stumbled upon this project when searching for gamparse and realized that it's not in the best of condition at this time. I played around with EQLogParser and am blown away by the interface and features. The perfect solution and answer to my prayers!

I play on an eqemu server, Wayfarer's Haven, which is at the GOD expansion. The server utilizes the spells_us.txt file. Right now no spell damage is showing up in the parser. Is there a possibility to have it possible to import / utilize the spells_us.txt file?

Thank you so much!

**Comment by kauffman12** [comment]
I'll have to look into what eqemu servers are doing. I imagine they're still using the log format from long ago. This parser only handles the format from the past couple years.

**Comment by AbyssalStalker** [comment]
Thanks for the reply!

I can put you in contact with the admin / dev of our server if you are looking for assistance with file identification. He's a really smart dude and contributes to the eqemu community.

My server is a few hundred strong and it's a great time. I showed this parser to my guild and they are foaming at the mouth hoping it can work for spells.

Cheers

**Comment by kauffman12** [comment]
Well you could just create a new log file and have your character cast some spells and have some else cast a few.  At least one of each for DD, DoT, Heal, etc and share that on here.

**Comment by AbyssalStalker** [comment]
[eqlog_Tatiana_Haven.txt](https://github.com/kauffman12/EQLogParser/files/7876234/eqlog_Tatiana_Haven.txt)

There's a log file that we use.

**Comment by AbyssalStalker** [comment]
Getting a parse for you momentarily. About to kill some mobs in a small raid

**Comment by AbyssalStalker** [comment]
[eqlog_Catenza_Haven.txt](https://github.com/kauffman12/EQLogParser/files/7876246/eqlog_Catenza_Haven.txt)

**Comment by kauffman12** [comment]
Interesting. I can probably make it parse the damage but since spell names aren't showing up for nukes you wouldn't be able to get spell counts. The damage summary, breakdown, damage log, and charts would work though. Definitely don't expect everything.

**Comment by fryguy503** [comment]
> Interesting. I can probably make it parse the damage but since spell names aren't showing up for nukes you wouldn't be able to get spell counts. The damage summary, breakdown, damage log, and charts would work though. Definitely don't expect everything.

I suggested to Abyssal that if the parser could utilize a servers spells_us.txt file it would be able to easily extrapolate what spells are being used.

e.g.  In Catenza's log example
Line 8500 - [Sat Jan 15 21:17:21 2022] Ptav M`sha is smothered in a rolling wave of flame.

[Long Date Time] (Target) (Cast On message)

Based on the spells_us.txt file that would be

1467^CallOfFireEffect^PLAYER_1^^^^A wave of flame washes over you.^ is smothered in a rolling wave of flame.^^100^0^0^0^0^0^0^0^0^0^0^0^-65^1^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^2513^2110^-1^-1^-1^-1^1^1^1^1^-1^-1^-1^-1^100^100^100^100^100^100^100^100^100^100^100^100^0^0^0^2^10^0^21^254^254^254^254^254^254^254^254^254^5^0^24^-1^0^0^255^255^255^255^255^255^255^255^255^255^255^255^255^255^255^255^43^0^0^18^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^100^0^161^84^0^-100^0^0^0^0^0^0^0^0^25^38^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^-100^100^0^-99^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^-1^0^0^0^1^0^0^1^1^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0^0

Which is Spell ID 1467 or "Call of Fire Effect" - Damage is -65 using Formula 100 (Effect Base Value)
Nearly all spell information can be inferred from this string, this would also make it universal for nearly all Emulated servers as they are required to keep all clients in sync with this file.

The spell file is just an entire list of carrot separated filed - https://docs.eqemu.io/server/spells/spell-effect-fields/

Example Emulated Spell File -- http://livepanel.wayfarershaven.com:3000/download/spells

**Comment by kauffman12** [comment]
I see where you going with that but I'm not sure how reliable it would be. On live servers it's gotten kinda crazy but I imagine some issues still apply here. 

The damage values are modified from focus effects. I'm not sure what expansion eqemu is based on but aren't there at least some SPA 124 abilities that modify base damage? Or crits for wizards. Or more than one spell having the same damage values or land on message.

 And another issue is the casting messages are visible based on distance to the player while the lands on messages are from distance to the NPC being hit. So, you may be close enough to see one message but not the other.  Also, don't some spells damage change based on level? Unless your version of EQ really most if not all those issues addressed it may not be very easy or accurate.

**Comment by fryguy503** [comment]
I agree, it won't be 100% accurate by any means. Was just trying to think of ways to improve data consistency from logs.

**Comment by kauffman12** [comment]
Ok. That makes sense. I'll add the basic parsing for now and maybe try something like that out a little later. I at least have one good log file to work with.

**Comment by kauffman12** [comment]
1.8.56 should work for simple parsing of direct damage, dots, and crits. I'll add the smarter parsing to a wish list but I wouldn't expect me to get to that very soon. Try that out and re-open if there's other simple bugs i can fix for parsing damage.
